petri/src/vm/mod.rs

```diff
@@ -68,6 +68,18 @@ pub trait PetriVmConfig: Send {
     /// Run the VM with Enable VMBus relay enabled
     fn with_vmbus_redirect(self: Box<Self>, enable: bool) -> Box<dyn PetriVmConfig>;
 
+    /// Apply custom configuration to the VM config (OpenVMM only)
```

Comment:
no, please refactor as much code as necessary to remove the "OpenVMM only" limitation